docs(Popover): Add documentation for `PopoverContent`, `PopoverFooter`, `PopoverHeader` & `PopoverLayout` (#2547)
looker-open-source/components: www/src/documentation/components/overlays/popoverlayout.mdx

---
title: PopoverLayout
storybook: true
---

> `PopoverLayout` is a container that has:

Reviewer: I'd take a look at `DialogLayout` for an example of the docs here:

```suggestion
The `PopoverLayout` component manages the popover layout components below to easily meet design-system expectations.

This component composes `PopoverHeader`, `PopoverContent` and `PopverFooter` as needed.
```

> `Header`: can be used as a label, scrolling inside a popover and with the ability to hava a close button. By default `Header` will display the closing button if `Fooret` is set to false.

Reviewer: `PopoverHeader`, `PopoverContent` & `PopoverFooter` should each have their own separate `mdx` files (see `Dialog` equivalents)

This section should get moved to `PopoverHeader` and so and so forth for the other sections.

```suggestion
`Header` can be used as a label, scrolling inside a popover and with the ability to have a a close button. By default `Header` will display the closing button if `Footer` is set to false.
```

`Content`: where the content will be with consistant spacing to help scrolling.

`Footer`: used to manage actions and scrolling inside a popover. `Footer` will be present unless specifically specified as false.
By default `Footer` will dispaly the work `Done` but that value can be changed to another string or component.

## Basic PopoverLayout

```jsx

> <PopoverLayout hideHeader header="Header">
>   PopoverLayout content
> </PopoverLayout>

Reviewer: I think the "Basic" example should probably show with a visible header as that's the recommended pratice.

```suggestion
<PopoverLayout header="Example Popover">
  PopoverLayout content
</PopoverLayout>
```

```

## PopoverLayout Header hidden

> 

Reviewer: This needs an explanation... e.g.: "Use the `hidden` prop to hide the visible header text but still provide context for screen readers"

```jsx
<PopoverLayout header="Header" footer={false}>
  PopoverLayout content
</PopoverLayout>
```

> ## PopoverLayout Footer extra content

Reviewer: Need to explain that `Done` is included by default and how to change the word "Done" to something else.

Author: we actually don't have this as an option right now. As of now you can only pass `footer` and that is a `ReactNode` so we are not specifying the close prop on `PopoverLayout` that may have to be an extra ticket if that should be included.

Reviewer: 🤦 . Yeah we should make that possible.

Reviewer: PR at #2557

Reviewer 2: ```suggestion
## Footer extra content
```

---

> title: PopoverLayout

Reviewer 2: Add github path:
```
github: components/src/Popover/PopoverLayout.tsx
```

storybook: true
---

The `PopoverLayout` component manages the popover layout components below to easily meet design-system expectations.

This component composes `PopoverHeader`, `PopoverContent` and `PopverFooter` as needed.

> `Header` will display the closing button if `Footer` is set to false.

Reviewer 2: ```suggestion
`PopoverHeader` will display the closing button if `footer` is set to false.
```

> `Footer`: By default `Footer` will dispaly the work `Done` but that value can be changed to another string or component.

Reviewer 2: ```suggestion
By default `PopoverFooter` will contain a dismissal button with the text "Done", but that value can be changed to another string or component.
```

> ## Basic PopoverLayout

Reviewer 2: I think this header is unnecessary.
```suggestion
```

```jsx
<PopoverLayout header="Example Popover">PopoverLayout content</PopoverLayout>
```

> ## PopoverLayout Header hidden

Reviewer 2: ```suggestion
## Header hidden
```

Use the hidden prop to hide the visible header text but still provide context for screen readers

```jsx
<PopoverLayout hideHeader header="Example Popover" footer={false}>
  PopoverLayout content
</PopoverLayout>
```

## PopoverLayout Footer extra content

Use the footer prop to add an extra element to the Footer.

```jsx
<PopoverLayout
  header="Header"
  footer={
    <ButtonTransparent color="neutral" size="small">
      Cancel
    </ButtonTransparent>
  }
>
  PopoverLayout content
</PopoverLayout>
```

> ## PopoverLayout no Footer

Reviewer 2: ```suggestion
## No footer
```
